security/pkg/caclient/protocol/protocol.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package protocol defines the interface of CA client protocol. Currently we only support gRPC
+// protocol sent to Istio CA server.
+package protocol
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	//"istio.io/istio/pkg/log"
+	"istio.io/istio/security/pkg/platform"
+	pb "istio.io/istio/security/proto"
+)
+
+// CAProtocol is the interface for talking to CA.
+type CAProtocol interface {
```

Comment:
Do we need to define this interface here?

Reply:
It's used in probe/controller.go and caclient/client.go, basically renamed & replaced existing GrpcClient. Any alternative suggestion?

Reply:
Resolved per offline discussion.